plugins/module_utils/version.py

```diff
@@ -10,4 +10,41 @@
 __metaclass__ = type
 
 
+import operator
+from functools import wraps
+
 from ansible.module_utils.compat.version import LooseVersion  # noqa: F401, pylint: disable=unused-import
+
+
+def _version_compare(op):
+    @wraps(op)
+    def _op(a, b):
+        return op(LooseVersion(a), LooseVersion(b))
+    return _op
+
+
+version_ops = {
+    '<=': _version_compare(operator.le),
+    '>=': _version_compare(operator.ge),
+    '<': _version_compare(operator.lt),
+    '>': _version_compare(operator.gt),
+    '==': _version_compare(operator.eq),
+    '!=': _version_compare(operator.ne),
+}
+
+
+def find_op(name):
+    if name is None:
+        return None, None, None
+    for op in version_ops:
```

Comment:
Note that the order is different for potentially every run on Python < 3.6, whence sometimes it will split `foo <= 3` as `foo `, `<`, `= 3`, and sometimes as `foo `, `<=`, ` 3`.